group_project_v2/stage_components.py

```diff
@@ -0,0 +1,551 @@
+        display_name=_(u"Question content"),
+        help=_(u"HTML control"),
+        default="",
+        scope=Scope.content,
+        multiline_editor="xml"
+    )
+
+    required = Boolean(
+        display_name=_(u"Required"),
+        default=False,
+        scope=Scope.content
+    )
+
+    grade = Boolean(
+        display_name=_(u"Grading"),
+        help=_(u"IF True, answers to this question will be used to calculate student grade for Group Project."),
+        default=False,
+        scope=Scope.content
+    )
+
+    single_line = Boolean(
+        display_name=_(u"Single line"),
+        help=_(u"If True question label and content will be displayed on single line, allowing for more compact layout."
+               u"Only affects presentation."),
+        default=False,
+        scope=Scope.content
+    )
+
+    question_css_classes = String(
+        display_name=_(u"CSS classes"),
+        help=_(u"CSS classes to be set on question element. Only affects presentation."),
+        scope=Scope.content
+    )
+
+    editable_fields = (
+        "question_id", "title", "assessment_title", "question_content", "required", "grade", "single_line",
+        "question_css_classes"
+    )
+
+    @lazy
+    def stage(self):
+        return self.get_parent()
+
+    def render_content(self):
+        try:
+            answer_node = ElementTree.fromstring(self.question_content)
+        except ElementTree.ParseError as exception:
+            log.exception(exception)
+            return ""
+
+        answer_node.set('name', self.question_id)
+        answer_node.set('id', self.question_id)
+        current_class = answer_node.get('class')
+        answer_classes = ['answer']
+        if current_class:
+            answer_classes.append(current_class)
+        if self.single_line:
+            answer_classes.append('side')
+        if self.stage.is_closed:
+            answer_node.set('disabled', 'disabled')
+        else:
+            answer_classes.append('editable')
+        answer_node.set('class', ' '.join(answer_classes))
+
+        return outer_html(answer_node)
+
+    def student_view(self, context):
+        question_classes = ["question"]
+        if self.required:
+            question_classes.append("required")
+        if self.question_css_classes:
+            question_classes.append(self.question_css_classes)
+
+        fragment = Fragment()
+        render_context = {
+            'question': self,
+            'question_classes': " ".join(question_classes),
+            'question_content': self.render_content()
+        }
+        render_context.update(context)
+        fragment.add_content(loader.render_template("templates/html/components/review_question.html", render_context))
+        return fragment
+
+    def studio_view(self, context):
+        fragment = super(GroupProjectReviewQuestionXBlock, self).studio_view(context)
+
+        # TODO: StudioEditableXBlockMixin should really support Codemirror XML editor
```

Comment:
@Kelketek no, I've hacked together a good-enough solution and I don' think I have time for this in scope of this task.